src/connector/postgres.rs

```diff
@@ -548,6 +548,78 @@ mod tests {
         assert_eq!(row["salary"].as_f64(), Some(20000.0));
     }
 
+    #[tokio::test]
+    async fn type_roundtrips() {
+        let table = r#"
+            CREATE TABLE types (
+                id SERIAL PRIMARY KEY,
+
+                bytes_uuid uuid,
+                network_inet inet,
+                numeric_float4 float4,
+                numeric_float8 float8,
+                time_timetz timetz,
+                time_time time,
+                time_date date,
+                text_jsonb jsonb,
+                time_timestamptz timestamptz
+            );
```

Comment:
we may be missing some here - I have more exhaustive tests in the query engine.

Also we're not testing arrays of these types, I'll need to confirm that they work.